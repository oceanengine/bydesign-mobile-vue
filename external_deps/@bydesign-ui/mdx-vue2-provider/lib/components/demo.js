/**
 * Component Demo
 */
import createBEM from '../utils/bem';
import HL from './highlight';
import { openCodepen, openSandbox, copyToClipboard } from '../utils/editor';
var bem = createBEM('demo');

var Demo = function Demo() {
  return {
    props: {
      title: [String, Number],
      desc: [String, Number],
      code: String,
      src: String,
      srcRender: Function,
      maxHeight: Number
    },
    data: function data() {
      return {
        unfolded: false
      };
    },
    methods: {
      setUnfolded: function setUnfolded(value) {
        this.unfolded = value;
      }
    },
    render: function render() {
      var h = arguments[0];
      var title = this.title,
          desc = this.desc,
          code = this.code,
          srcRender = this.srcRender,
          unfolded = this.unfolded,
          setUnfolded = this.setUnfolded,
          _this$maxHeight = this.maxHeight,
          maxHeight = _this$maxHeight === void 0 ? 200 : _this$maxHeight;
      var containerStyle = {
        maxHeight: "".concat(maxHeight, "px")
      };
      return h("div", {
        "class": bem()
      }, [h("div", {
        "class": bem('title'),
        "attrs": {
          "id": title.toString()
        }
      }, [title]), srcRender !== undefined && h("div", {
        "class": bem('box')
      }, [h("div", {
        "class": bem('show')
      }, [srcRender()]), desc && h("div", {
        "class": bem('desc')
      }, [desc]), code !== undefined && h("div", {
        "class": bem('code-tool')
      }, [h("div", {
        "class": bem('operations')
      }, [h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openCodepen(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/codepen.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openSandbox(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/sandbox.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            copyToClipboard(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/copy.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            setUnfolded(!unfolded);
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/code.svg')
        }
      })])]), code !== undefined && unfolded && h("div", {
        "class": bem('code'),
        "style": containerStyle
      }, [h(HL, {
        "attrs": {
          "code": unescape(code)
        }
      })])])]);
    }
  };
};

export default Demo;