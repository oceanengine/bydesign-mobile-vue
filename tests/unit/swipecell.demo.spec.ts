import Demo from '@src/components/swipe-cell/demo/index.vue';
import { snapshotDemo } from '../demo';

snapshotDemo(Demo);
