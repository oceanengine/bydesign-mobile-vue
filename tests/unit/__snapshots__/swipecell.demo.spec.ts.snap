// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo swipe-cell">
  <div class="byted-header--wrapper">
    <div class="byted-header byted-header--default byted-header--fixed byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title"><span class="demo-header-icon"><img src="https://s2.pstatp.com/cdn/expire-1-d/byted-creative-app/static/ByDesign-logo.svg" class="img-icon"></span></div>
    </div>
  </div>
  <div style="padding-left: 0.8rem;"><span class="byted-icon byted-icon-hand-drag" style="font-size: 20px;"><svg width="20" height="20" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M9.58291 27.1824C7.86707 28.3542 7.00916 30.2934 7.00916 33.0002C7.00916 37.0602 12 44.0002 16.5004 44.0002C21.0009 44.0002 23.6109 44.0002 28.0159 44.0002C32.4209 44.0002 35.0964 40.1495 35.0964 37.0602C35.0964 32.9069 35.0964 28.7536 35.0964 24.6002C35.0964 22.8072 33.6454 21.3522 31.8524 21.3472C30.0658 21.3422 28.6134 22.7865 28.6084 24.5731C28.6084 24.5761 28.6084 24.5791 28.6084 24.5821V24.6836" stroke="#333" stroke-width="4" stroke-linecap="round"></path><path d="M10.9811 29.4453V7.66246C10.9811 5.88568 12.4214 4.44531 14.1982 4.44531C15.975 4.44531 17.4154 5.88568 17.4154 7.66246V23.6479" stroke="#333" stroke-width="4" stroke-linecap="round"></path><path d="M17.4154 24.0001V19.8076C17.4154 18.2589 18.6709 17.0034 20.2196 17.0034C21.7683 17.0034 23.0238 18.2589 23.0238 19.8076V24.4272" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M23 24.6583V21.8076C23 20.2589 24.2555 19.0034 25.8042 19.0034C27.3529 19.0034 28.6084 20.2589 28.6084 21.8076V25.0034" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M11 8H41" stroke="#333" stroke-width="4" stroke-linecap="round"></path><path d="M36 12.5L37.6667 11L41 8L37.6667 5L36 3.5" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path></svg></span></div>
  <div>
    <div class="swipe-cell-content">
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-swipe-cell__left">
            <div class="byted-button byted-button--primary byted-button--normal byted-button--square"></div>
          </div>
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
        </div>
      </div>
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-swipe-cell__left">
            <div class="byted-button byted-button--primary byted-button--normal byted-button--square"></div>
            <div class="byted-button byted-button--danger byted-button--normal byted-button--square"></div>
          </div>
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
        </div>
      </div>
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
          <div class="byted-swipe-cell__right">
            <div class="byted-button byted-button--primary byted-button--normal byted-button--square"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="swipe-cell-content">
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-swipe-cell__left">
            <div style="width: 40px; height: 100%; display: flex; align-items: center; justify-content: center;"><i class="byted-icon byted-icon--check" style="fill: #666; width: 24px; height: 24px;">test-file-stub</i></div>
          </div>
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
        </div>
      </div>
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-swipe-cell__left">
            <div class="left-two">
              <div class="icon-box"><i class="byted-icon byted-icon--check" style="fill: #666; width: 24px; height: 24px;">test-file-stub</i></div>
              <div class="icon-box"><i class="byted-icon byted-icon--arrow" style="fill: #666; width: 24px; height: 24px;">test-file-stub</i></div>
            </div>
          </div>
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
        </div>
      </div>
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
          <div class="byted-swipe-cell__right">
            <div style="width: 40px; height: 100%; display: flex; align-items: center; justify-content: center;"><i class="byted-icon byted-icon--check" style="fill: #666; width: 24px; height: 24px;">test-file-stub</i></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="swipe-cell-content">
      <div class="byted-swipe-cell">
        <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
          <div class="byted-cell">
            <div class="byted-cell__title"><span></span></div>
            <div class="byted-cell__value byted-cell__value--alone"><span></span></div>
          </div>
          <div class="byted-swipe-cell__right">
            <div class="byted-button byted-button--danger byted-button--normal byted-button--square"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
